soc/intel/intel_adsp/ace/power.c

```diff
@@ -340,7 +340,11 @@ void pm_state_set(enum pm_state state, uint8_t substate_id)
 			sys_cache_data_flush_range((void *)imr_layout, sizeof(*imr_layout));
 #endif /* CONFIG_ADSP_IMR_CONTEXT_SAVE */
 			/* do power down - this function won't return */
-			power_down(true, CONFIG_ADSP_POWER_DOWN_HPSRAM, true);
+#ifdef CONFIG_ADSP_POWER_DOWN_HPSRAM
+			power_down(true, true, true);
+#else
+			power_down(true, false, true);
+#endif /* CONFIG_ADSP_POWER_DOWN_HPSRAM */
```

Comment:
would `power_down(true, IS_ENABLED(CONFIG_ADSP_POWER_DOWN_HPSRAM), true);` work?

Reply:
duh, yes, of course. I know it looked uglier than it should have... Changed, thanks!